test/tests/functional/pbs_init_script.py

```diff
@@ -71,6 +71,22 @@ def test_env_vars_precede_pbs_conf_file(self):
         self.assertNotIn("PBS comm", output)
         self.assertIn("PBS mom", output)
 
+    def test_init_services_pid(self):
+        """
+        Test if the pid of PBS daemons are updated correctly
+        after a restart via PBSInitServices's functions
+        """
+        self.server.pi.stop()
+        self.server.pi.start_server()
+        self.server.pi.start_mom()
+        self.server.pi.start_comm()
+        self.server.pi.start_sched()
+
+        self.assertTrue(self.server.signal('-HUP'))
+        self.assertTrue(self.mom.signal('-HUP'))
+        self.assertTrue(self.scheduler.signal('-HUP'))
+        self.assertTrue(self.comm.signal('-HUP'))
+
```

Comment:
Should this be here or in self tests?

Reply:
Yes. It makes more sense under selftest dir. I have added a new file in the selftest dir, as none of the existing files were suitable for this.